Source/JavaScriptCore/wasm/WasmSlowPaths.cpp

```diff
@@ -191,11 +191,14 @@ static inline bool jitCompileSIMDFunction(Wasm::LLIntCallee* callee, Wasm::Insta
             tierUpCounter.m_compilationStatus = Wasm::LLIntTierUpCounter::CompilationStatus::Compiling;
             break;
         case Wasm::LLIntTierUpCounter::CompilationStatus::Compiling:
-            // This spinlock is bad, but this is only temporary.
+            // FIXME: This spinlock is bad, but this is only temporary. https://bugs.webkit.org/show_bug.cgi?id=274326
             continue;
         case Wasm::LLIntTierUpCounter::CompilationStatus::Compiled:
-            RELEASE_ASSERT(!!callee->replacement(instance->memory()->mode()));
-            return true;
+            if (!!callee->replacement(instance->memory()->mode()))
+                return true;
+            callee->clearOtherReplacementsIfExisting(instance->memory()->mode());
+            tierUpCounter.m_compilationStatus = Wasm::LLIntTierUpCounter::CompilationStatus::NotCompiled;
```

Comment:
I think this looks strange. LLInt code is shared by both memory mode, but BBQ / OMG are not. And if we get compilation status from tierUpCounter, this is still unsure whether this is compilation status for Signaling mode / BoundChecking mode.
I think the fix is having Signaling / BoundChecking per-mode status in LLIntTierUpCounter

Reply:
Changed.